x-pack/plugins/security_solution/public/common/components/error_toast_dispatcher/index.tsx

```diff
@@ -4,25 +4,24 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { useEffect } from 'react';
-import { connect, ConnectedProps } from 'react-redux';
+import React, { useEffect, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
 
-import { appSelectors, State } from '../../store';
+import { useDeepEqualSelector } from '../../hooks/use_selector';
+import { appSelectors } from '../../store';
 import { appActions } from '../../store/app';
 import { useStateToaster } from '../toasters';
 
 interface OwnProps {
   toastLifeTimeMs?: number;
 }
 
-type Props = OwnProps & PropsFromRedux;
-
-const ErrorToastDispatcherComponent = ({
-  toastLifeTimeMs = 5000,
-  errors = [],
-  removeError,
-}: Props) => {
+const ErrorToastDispatcherComponent: React.FC<OwnProps> = ({ toastLifeTimeMs = 5000 }) => {
+  const dispatch = useDispatch();
+  const getErrorSelector = useMemo(() => appSelectors.errorsSelector(), []);
+  const errors = useDeepEqualSelector(getErrorSelector);
   const [{ toasts }, dispatchToaster] = useStateToaster();
+
   useEffect(() => {
```

Comment:
I am confuse here, why we do not have any dependency on the useEffect. Did you try to add `[dispatch, dispatchToaster, errors, toastLifeTimeMs, toasts]`